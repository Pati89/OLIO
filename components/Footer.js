import styled from "@emotion/styled";

const S_footer = styled.footer`
  display: flex;
  flex: 1;
  padding: 2rem 0;
  border-top: 1px solid #eaeaea;
  justify-content: center;
  align-items: center;
`;

const S_span_bold = styled.span`
  font-weight: bold;
`;

const Footer = () => {
  return (
    <S_footer>
      <p>
        Developed by <S_span_bold>Patrycja Banaszczyk</S_span_bold>
      </p>
    </S_footer>
  );
};

export default Footer;
