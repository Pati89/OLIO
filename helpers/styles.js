// here we can add theme, fontSizes or spacing sizes to use same all over the page

export const breakpoints = {
  sm: 640,
  md: 896,
  lg: 1092,
};

export const media = {
  /* small size and up */
  sm: `@media only screen and (min-width: ${breakpoints.xs}px)`,
  /* medium size and up */
  md: `@media only screen and (min-width: ${breakpoints.sm}px)`,
  /* large size and up */
  lg: `@media (min-width: ${breakpoints.md}px)`,
};
