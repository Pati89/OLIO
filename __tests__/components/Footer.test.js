import Footer from "../../components/Footer";
import { render } from "../../helpers/test-utils";

describe("Footer", () => {
  it("should render a Footer", () => {
    const { getByTestId, getByText } = render(<Footer />);

    expect(getByTestId("footer")).toBeTruthy();
    expect(getByText("Developed by")).toBeTruthy();
    expect(getByText("Patrycja Banaszczyk")).toHaveStyleRule(
      "font-weight",
      "bold"
    );
  });

  it("should render a Footer match snapshot", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
