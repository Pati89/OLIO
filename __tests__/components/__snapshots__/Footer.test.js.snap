// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render a Footer match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 2rem 0;
  border-top: 1px solid #eaeaea;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  font-weight: bold;
}

<div>
  <footer
    class="emotion-0"
    data-testid="footer"
  >
    <p>
      Developed by 
      <span
        class="emotion-1"
      >
        Patrycja Banaszczyk
      </span>
    </p>
  </footer>
</div>
`;
